### PR TITLE
FS-1009 feat config loader

### PR DESCRIPTION
**Pull request by SentientHousePlant**
Adds a pretty printing method to config classes.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+[tool.poetry.dependencies]
+python = "^3.10"
+python-dotenv = "^0.20.0"
+PyYAML = "^6.0"
+rich = "^12.4.4"
+
+[tool.poetry.dev-dependencies]
+
+[build-system]
+requires = ["poetry-core>=1.0.0"]
```
Comment (gidsg) [inline pyproject.toml:17]: Do we use poetry elsewhere? What's the thinking in introducing it here?

File: setup.py [hunk 2]
```diff
@@ -5,20 +5,26 @@
...
+    "rich>=12.4.4,<13.0.0",
+]
+
 setup_kwargs = {
     "name": "funding-service-design-utils",
-    "version": "0.0.3",
+    "version": "0.0.4",
     "description": "Utils for the fsd-tech team",
     "long_description": None,
-    "author": "DLUHC",
```
Comment (gio-karunakaran) [inline setup.py:13]: 🤔 I think this is right and should be DLUHC?

File: fsd_tech/configclass.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    return cls
+
+
+if __name__ == "__main__":
+
+    @configclass
+    class Default:
+
+        my_val = 1
```
Comment (gio-karunakaran) [inline fsd_tech/configclass.py:58]: 🤔 is this testing stuff that can be removed?

**Review by gio-karunakaran — COMMENTED** [review]
just a couple queries! but looks great other than that! :)

**Comment by SentientHousePlant** [comment]
@gidsg fsd_utils is preferred because that will be the name of the imported module

File: fsd_tech/configclass.py (hunk 3 above)
Comment (SentientHousePlant) [reply fsd_tech/configclass.py:58]: Hm, its useful for the dev working on it because we cannot easily test the formatting of printing.

Comment (SentientHousePlant) [reply fsd_tech/configclass.py:58]: But I'll relinquish it to the void.

**Review by gio-karunakaran — APPROVED** [review]
LGTM :)

File: fsd_utils/configclass.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    for k, v in cls.__dict__.items():
+        if not (k.startswith("__") or k.endswith("__")):
+            settled_bases_config_info[k] = v
+
+    # This is the dicitonary that will be used
+    #  by classes which inherit this class
+    cls._config_info_ = settled_bases_config_info
+
+    @classmethod
+    def pretty_print(self):
```
Comment (srh-sloan) [inline fsd_utils/configclass.py:29]: In future we should allow this to work with the logging utils so any pretty printing is written to logs not console

**Review by srh-sloan — CHANGES_REQUESTED** [review]
Can we add a README (new or section on existing) about how to use this in a real app please?

File: fsd_utils/configclass.py (hunk 4 above)
Comment (SentientHousePlant) [reply fsd_utils/configclass.py:29]: Sounds good :) I'll create a BAU pr once harry's stuff has been added.

File: setup.py [hunk 5]
```diff
@@ -5,20 +5,26 @@
...
+    "author": "DHULC",
```
Comment (srh-sloan) [inline setup.py:19]: It should be DLUHC

**Review by srh-sloan — APPROVED** [review]

**Pull request merged**